packages/permissionless/errors/account.ts

```diff
@@ -0,0 +1,306 @@
+                "",
+                "Possible solutions:",
+                "• If this is the first transaction by this account, make sure the initCode is included in the user operation.",
+                "• If the smart account is already supposed to be deployed, verify that you have selected the correct sender address for the user operation.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountDoNotHaveEnoughFunds extends BaseError {
+    static message = /aa21/
+    override name = "SmartAccountDoNotHaveEnoughFunds"
+    constructor({
+        cause,
+        sender,
+        docsPath
+    }: {
+        cause?: BaseError
+        sender: Address
+        docsPath?: string
+    }) {
+        super(
+            [
+                `You are not using a paymaster, but the ${sender} address did not have enough native tokens to cover the gas costs associated with the user operation.`,
+                "",
+                "Possible solutions:",
+                "• If you are not using a paymaster, verify that the sender address has enough native tokens to cover the required pre-fund. Consider leveraging functions like getRequiredPrefund.",
+                "• If you are looking to use a paymaster to cover the gas fees, verify that the paymasterAndData field is set.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountSignatureExpiredOrNotDue extends BaseError {
+    static message = /aa21/
+    override name = "SmartAccountDoNotHaveEnoughFunds"
+    constructor({
+        cause,
+        docsPath
+    }: {
+        cause?: BaseError
+        docsPath?: string
+    }) {
+        super(
+            [
+                "The signature used in the user operation is not valid, because it is outside of the time range it specified.",
+                "",
+                "Possible reasons:",
+                "• This error occurs when the block.timestamp falls after the validUntil timestamp, or before the validAfter timestamp.",
+                "",
+                "Possible solutions:",
+                "• If you are looking to use time-based signatures, verify that the validAfter and validUntil fields are set correctly and that the user operation is sent within the specified range.",
+                "• If you are not looking to use time-based signatures, verify that the validAfter and validUntil fields are set to 0.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountRevertedOrOutOfGasDuringValidation extends BaseError {
```

Comment:
SmartAccountRevertedOrOutOfGasDuringValidation -> ValidationRevertedOrOutOfGas?

Reply:
Can do this, but since we have SmartAccount in front of everything I though let's do this too=